packages/adapters-library/src/adapters/stargate/products/farm-v2/stargateFarmV2Adapter.ts

```diff
@@ -0,0 +1,374 @@
+
+    const lpStakingContract = StargateStaking__factory.connect(
+      stargateStakingAddress,
+      this.provider,
+    )
+
+    const filter =
+      filterType === 'deposit'
+        ? lpStakingContract.filters.Deposit(
+            protocolTokenAddress,
+            undefined,
+            userAddress, // Address that receives the tokens (to)
+            undefined,
+          )
+        : lpStakingContract.filters.Withdraw(
+            protocolTokenAddress,
+            userAddress, // Address from which the tokens are withdrawn (from)
+            undefined,
+            undefined,
+          )
+
+    const events = await lpStakingContract.queryFilter(
+      filter,
+      fromBlock,
+      toBlock,
+    )
+
+    return events.map((event) => {
+      const { amount } = event.args!
+
+      return {
+        protocolToken: {
+          address: protocolToken.address,
+          name: protocolToken.name,
+          symbol: protocolToken.symbol,
+          decimals: protocolToken.decimals,
+        },
+        blockNumber: event.blockNumber,
+        transactionHash: event.transactionHash,
+        tokens: [
+          {
+            address: protocolToken.address,
+            name: protocolToken.name,
+            symbol: protocolToken.symbol,
+            decimals: protocolToken.decimals,
+            type: TokenType.Underlying,
+            blockNumber: event.blockNumber,
+            balanceRaw: amount,
+          },
+        ],
+      }
+    })
+  }
+
+  async getRewardWithdrawals({
+    userAddress,
+    protocolTokenAddress,
+    fromBlock,
+    toBlock,
+  }: GetEventsInput): Promise<MovementsByBlock[]> {
+    const { rewardTokens, rewarderAddress, address, symbol, name, decimals } =
+      await this.getProtocolTokenByAddress(protocolTokenAddress)
+
+    const multiRewarderContract = StargateMultiRewarder__factory.connect(
+      rewarderAddress,
+      this.provider,
+    )
+
+    const filter = multiRewarderContract.filters.RewardsClaimed(
+      userAddress,
+      undefined,
+      undefined,
+    )
+
+    const events = await multiRewarderContract.queryFilter(
+      filter,
+      fromBlock,
+      toBlock,
+    )
+
+    return events.map((event) => {
+      const { rewardTokens: rewardTokenAddresses, amounts } = event.args!
+
+      return {
+        protocolToken: {
+          address,
+          name,
+          symbol,
+          decimals,
+        },
+        blockNumber: event.blockNumber,
+        transactionHash: event.transactionHash,
+        tokens: rewardTokenAddresses.map((rewardTokenAddress, i) => {
+          const rewardAmount = amounts[i]!
+
+          const rewardToken = rewardTokens.find(
+            (rewardToken) => rewardToken.address === rewardTokenAddress,
+          )!
+
+          return {
+            address: rewardToken.address,
+            name: rewardToken.name,
+            symbol: rewardToken.symbol,
+            decimals: rewardToken.decimals,
+            type: TokenType.Underlying,
```

Comment:
underlying-claimable

Reply:
As mentioned before. I'm using the same type that other `getRewardWithdrawals` implementations (curve and convex) use.

Maybe we need to change them all.